presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/PlanBuilder.java

```diff
@@ -86,6 +91,29 @@ public ApplyNode apply(Assignments subqueryAssignments, List<Symbol> correlation
         return new ApplyNode(idAllocator.getNextId(), input, subquery, subqueryAssignments, correlation);
     }
 
+    public TableWriterNode tableWriter(
+            PlanNode source,
+            TableWriterNode.WriterTarget target,
+            List<Symbol> columns,
+            List<String> columnNames,
+            List<Symbol> outputs,
+            Optional<PartitioningScheme> partitioningScheme)
+    {
+        return new TableWriterNode(
+                idAllocator.getNextId(),
+                source,
+                target,
+                columns,
+                columnNames,
+                outputs,
+                partitioningScheme);
+    }
+
+    public UnionNode union(List<PlanNode> sources, ListMultimap<Symbol, Symbol> outputsToInputs, List<Symbol> outputs)
```

Comment:
you don't need to get outputs here, as I understand they can be calculated from outputsToInputs